crates/optimism/primitives/src/supervisor.rs

```diff
@@ -0,0 +1,47 @@
+//! This is our custom implementation of validator struct
+
+use alloy_rpc_client::ReqwestClient;
+pub use kona_interop::{ExecutingDescriptor, SafetyLevel};
+pub use kona_rpc::{
+    CheckAccessList, InteropTxValidator, InteropTxValidatorError, SupervisorApiClient,
+};
+use std::time::Duration;
+
+/// Implementation of the supervisor trait for the interop.
+#[derive(Debug, Clone)]
+pub struct SupervisorClient {
+    inner: kona_rpc::SupervisorClient,
```

Comment:
all this type does is send one single request:

https://github.com/op-rs/kona/blob/0d4886ace24afc9e51692cba7ee26ec29f73fe0b/crates/node/rpc/src/reqwest.rs#L24-L28

imo we can do all of this without pulling in the kona client dependency here

either via alloy's provider if we need websocket and HTTP support.

or we can use reqwest directly:

https://github.com/paradigmxyz/reth/blob/9cb7177dfc692111888068f52b9b89db0fce2177/crates/optimism/rpc/src/sequencer.rs#L66-L75

Reply:
There is also function to parse access list, but it could be taken into the reth too